ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java

```diff
@@ -224,63 +232,122 @@ public void addHostToCluster(@EffectorParam(name = "Cluster name") String cluste
+                              @EffectorParam(name = "Stack Name") String stackName,
+                              @EffectorParam(name = "Stack version") String stackVersion,
+                              @EffectorParam(name = "Hosts", description = "List of FQDNs to add to cluster") List<String> hosts,
+                              @EffectorParam(name = "Services", description = "List of services to install on cluster") List<String> services,
+                              @EffectorParam(name = "Configurations", description = "Map of configurations to apply to blueprint") Map<String, Map> config) {
+        final RecommendationWrappers recommendationWrappers = getRecommendations(stackName, stackVersion, hosts, services);
+
+        deployCluster(clusterName, blueprintName, recommendationWrappers.getRecommendationWrappers().size() > 0 ? recommendationWrappers.getRecommendationWrappers().get(0) : null, config);
     }
 
     @Override
-    public void createComponentToCluster(@EffectorParam(name = "Cluster name") String cluster,
-                                         @EffectorParam(name = "Service name") String service,
-                                         @EffectorParam(name = "Component name") String component) {
+    public void updateStackRepository(@EffectorParam(name = "Stack Name") String stackName, @EffectorParam(name = "Stack Version") String stackVersion, @EffectorParam(name = "Operating System") String os, @EffectorParam(name = "Repository Name") String repoName, @EffectorParam(name = "Repository URL") String url) {
         waitForServiceUp();
-        restAdapter.create(ServiceEndpoint.class).createComponent(cluster, service, component);
+        restAdapter.create(StackEndpoint.class)
+                .updateStackRepository(stackName, stackVersion, os, repoName, ImmutableMap.builder()
+                        .put("Repositories", ImmutableMap.builder()
+                                .put("base_url", url)
+                                .put("verify_base_url", true)
+                                .build())
+                        .build());
     }
 
     @Override
-    public void createHostComponent(@EffectorParam(name = "Cluster name") String cluster,
-                                    @EffectorParam(name = "Host FQDN") String hostName,
-                                    @EffectorParam(name = "Component name") String component) {
+    public void addServiceToCluster(@EffectorParam(name = "Cluster name") final String cluster,
+                                    @EffectorParam(name = "Service name") final String service,
+                                    @EffectorParam(name = "Mappings of component to host") Map<String, String> mappings,
+                                    @EffectorParam(name = "Service configuration key", nullable = true, defaultValue = EffectorParam.MAGIC_STRING_MEANING_NULL) String configurationKey,
```

Comment:
Yes, that's why I was suggesting to replace parameters `configurationKey` and `configuration` by only one parameter `Map<String, Map<String, String>> configuration`